library/data/registry-entry-attributes/src/testmod/java/org/quiltmc/qsl/registry/attributes/test/DispatchedAttributeTest.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2021 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.registry.attributes.test;
+
+import com.google.common.collect.ImmutableMap;
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+import net.fabricmc.api.ModInitializer;
+
+import org.quiltmc.qsl.registry.attribute.api.DispatchedType;
+import org.quiltmc.qsl.registry.attribute.api.RegistryEntryAttribute;
+import org.quiltmc.qsl.registry.attribute.api.RegistryExtensions;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.text.LiteralText;
+import net.minecraft.text.Text;
+import net.minecraft.util.*;
+import net.minecraft.util.registry.Registry;
+import net.minecraft.world.World;
+import java.util.Map;
+
+public class DispatchedAttributeTest implements ModInitializer {
```

Comment:
Hmm this is definitely an interesting use case. I feel the docs in api would be much nicer given this example and explaining how dispatched types would allow for functionally composable items, blocks, block entities, etc.